rider-core/src/main/java/com/github/database/rider/core/dataset/DataSetExecutorImpl.java

```diff
@@ -861,7 +883,90 @@ public void compareCurrentDataSetWith(DataSetConfig expectedDataSetConfig, Strin
+        StringBuilder sbFacts = new StringBuilder();
+        List<String> queryTerms = new ArrayList<>();
+
+        for (String tableName : tableNames) {
+            ITable expectedTable = null;
+            ITable actualTable = null;
+            try {
+                expectedTable = expected.getTable(tableName);
+                actualTable = current.getTable(tableName);
+            } catch (DataSetException e) {
+                throw new RuntimeException("DataSet comparison failed due to following exception: ", e);
+            }
 
+            for (int i = 0; i < actualTable.getRowCount(); i++) {
+                sbFacts.append(actualTable.getTableMetaData().getTableName().toLowerCase());
+                sbFacts.append("(");
+                ITable finalActualTable = actualTable;
+                int finalI = i;
+                sbFacts.append(
+                        Arrays.stream(actualTable.getTableMetaData().getColumns())
+                                .map(column -> {
+                                    try {
+                                        return finalActualTable.getValue(finalI, column.getColumnName());
+                                    } catch (DataSetException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                })
+                                .map(o -> "'" + o + "'")
+                                .collect(Collectors.joining(","))
+                );
+                sbFacts.append(").\n");
+            }
+
+            for (int i = 0; i < expectedTable.getRowCount(); i++) {
+                StringBuilder sbQuery = new StringBuilder();
+                sbQuery.append(expectedTable.getTableMetaData().getTableName().toLowerCase());
+                sbQuery.append("(");
+                int finalI = i;
+                ITable finalExpectedTable = expectedTable;
+                sbQuery.append(
+                        Arrays.stream(actualTable.getTableMetaData().getColumns())
+                                .map(column -> {
+                                    try {
+                                        return finalExpectedTable.getValue(finalI, column.getColumnName());
+                                    } catch (DataSetException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                })
+                                // FIXME: simple #toString() may not work well with all datatypes
+                                .map(o -> o == null ? "_" : (o.toString().startsWith("$$") && o.toString().endsWith("$$")) ? o.toString().replaceAll("\\$\\$", "").toUpperCase() : "'" + o + "'")
+                                .collect(Collectors.joining(","))
+                );
+                sbQuery.append(")");
+                queryTerms.add(sbQuery.toString());
+            }
```

Comment:
Can we extract to something like `createQueryTerms(queryTerms, expectedTable, actualTable, i);`?

I think we can follow the same idea of tableFacts and return a String in the createQueryTerm:

```
 for (int i = 0; i < expectedTable.getRowCount(); i++) {
                queryTerms.add(createQueryTerm(expectedTable, actualTable, i));
            }
```